### PR TITLE
Use regex to catch ignored paths instead of enumerating directories

### LINKED ISSUE #510: Files are not added to the Queue, no images are classified
**Describe the bug**
The app just does nothing at all. There are many images on this instance, but none of them get added to the queue. Reinstalling the app does not fix the issue. Initially, about 30 Photos were classified, then it just stopped doing anything. Running all the available occ recognize commands also does not help.

**To Reproduce**
1. Install Recognize
2. Wait for Pictures to be added to the queue
3. nothing happens

**Expected behavior**
Images get recognized

**Recognize (please complete the following information):**
 - Installed app version: v3.2.2
 - WASM mode: No
 - Enabled modes: face recognition

**Server (please complete the following information):**
 - Nextcloud version: 25.0.1
 - OS: Ubuntu 22.04
 - RAM: 16GB
 - Processor Architecture: x86_64

**Additional context**
App config:
`"recognize": {
            "enabled": "yes",
            "faces.batchSize": "5",
            "faces.enabled": "true",
            "faces.lastFile": "1668711743",
            "faces.status": "true",
            "imagenet.batchSize": "100",
            "imagenet.enabled": "false",
            "installed_version": "3.2.2",
            "landmarks.batchSize": "100",
            "landmarks.enabled": "false",
            "movinet.batchSize": "5",
            "movinet.enabled": "false",
            "movinet.lastFile": "1668720710",
            "movinet.status": "true",
            "musicnn.batchSize": "100",
            "musicnn.enabled": "false",
            "node_binary": "\/var\/www\/nextcloud\/apps\/recognize\/bin\/node",
            "tensorflow.cores": "0",
            "tensorflow.gpu": "false",
            "tensorflow.purejs": "false",
            "types": "filesystem,dav"
        }`

The tables `[oc_recognize_face_clusters, oc_recognize_face_detections, oc_recognize_queue_faces, oc_recognize_queue_imagenet, oc_recognize_queue_landmarks, oc_recognize_queue_movinet, oc_recognize_queue_musicnn]` are all empty.

### PR DESCRIPTION
**Pull request by marcelklehr**
~~depends on #497~~
fixes #510

### CODE REVIEW

File: lib/Service/IgnoreService.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
+
```
Comment (come-nc) [inline lib/Service/IgnoreService.php:2]: License header?

File: lib/Service/IgnoreService.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		$this->systemConfig = $systemConfig;
+		$this->logger = $logger;
+	}
+	/**
+	 * @param int $storageId
+	 * @param array $ignoreMarkers
+	 * @return list<string>
+	 */
+	public function getIgnoredDirectories(int $storageId, array $ignoreMarkers): array {
+		$qb = new CacheQueryBuilder($this->db, $this->systemConfig, $this->logger);
```
Comment (come-nc) [inline lib/Service/IgnoreService.php:27]: What is CacheQueryBuilder?
Is the call to new necessary? (`$OC::server->get(CacheQueryBuilder::class)` should work?)

Applications should not use stuff from `OC`

Comment (marcelklehr) [reply lib/Service/IgnoreService.php:27]: CacheQueryBuilder is an enhanced QueryBuilder class to access the file cache table. There is no OCP API for this, sadly, but we need to use the filecache table for performance reasons. We also need to create a new CacheQueryBuilder for each query, so I think $OC::server->get doesn't work.

File: lib/Service/IgnoreService.php (hunk 1 above)
Comment (marcelklehr) [reply lib/Service/IgnoreService.php:2]: :+1:

**Review by come-nc — APPROVED** [review]

**Pull request merged**